lib/core/scheduler.rb

```diff
@@ -195,5 +148,67 @@ def connector_registered?(service_type)
         false
       end
     end
+
+    def schedule_triggered?(scheduling_settings, identifier, last_synced)
+      # Don't sync if sync is explicitly disabled
+      unless scheduling_settings.present? && scheduling_settings[:enabled] == true
+        Utility::Logger.debug("#{identifier.capitalize} scheduling is disabled.")
+        return false
+      end
+
+      current_schedule = scheduling_settings[:interval]
+
+      # Don't sync if there is no actual scheduling interval
+      if current_schedule.nil? || current_schedule.empty?
+        Utility::Logger.warn("No sync schedule configured for #{identifier}.")
+        return false
+      end
+
+      current_schedule =
+        begin
+          Utility::Cron.quartz_to_crontab(current_schedule)
+        rescue StandardError => e
+          Utility::ExceptionTracking.log_exception(e, "Unable to convert quartz (#{current_schedule}) to crontab.")
+          return false
+        end
+      cron_parser = Fugit::Cron.parse(current_schedule)
+
+      # Don't sync if the scheduling interval is non-parsable
+      unless cron_parser
+        Utility::Logger.error("Unable to parse sync schedule for #{identifier}: expression #{current_schedule} is not a valid Quartz Cron definition.")
+        return false
+      end
+
+      # We want to sync when sync never actually happened
+      if last_synced.nil? || last_synced.empty?
```

Comment:
Does this mean that when I configure a new custom schedule it will be triggered during the next service iteration because `last_syncted` is `nil`? 

if yes, I don't think customers expect this behavior.

Reply:
Default schedules work this way so I don't think it's unprecedented behaviour.

If we don't trigger this when `last_synced: nil` then there's no way to determine if the next sync is ready because `last_synced` won't get generated by the first sync.
If we had a UI or creation flow we could force a default of `last_synced: Time.now` but that won't be available in 8.7.
🤔

Would you be comfortable with this if it was well-documented?

Reply:
> Would you be comfortable with this if it was well documented?

I wouldn't :). I think this is a bad customer experience. When I schedule a job to be triggered at 6:00pm I expect that it will run at 6:00pm and not right after I clicked `Save`. 

I would discuss this question with PMs.

Reply:
@vidok thanks for bringing this up as it has sparked a lot of conversation around how connector scheduling should function if the connector has never synced before.

I've just learnt that the connectors cron logic requires a `last_synced` value in order to calculate when the next sync should run. Because of that, we need to populate `last_synced` with a value in order to trigger any schedules. Until this is resolved I think we should implement custom scheduling with the same requirements that regular scheduling has.

[Relevant connectors code](https://github.com/elastic/connectors-ruby/blob/9d5be7a79ee18c794403d14b1dcfc2c574786771/lib/core/scheduler.rb#L122)
[Slack thread](https://elastic.slack.com/archives/C01795T48LQ/p1674644501487809)

Reply:
There's also an active bug issue for tracking this here: https://github.com/elastic/enterprise-search-team/issues/3366